pyscriptjs/src/plugins/pyterminal.ts

```diff
@@ -1,28 +1,19 @@
 import type { PyScriptApp } from '../main';
 import type { AppConfig } from '../pyconfig';
-import { Plugin } from '../plugin';
-import { UserError, ErrorCode } from '../exceptions';
+import { Plugin, checkedConfigOption, validateConfigParameter } from '../plugin';
 import { getLogger } from '../logger';
 import { type Stdio } from '../stdio';
 import { InterpreterClient } from '../interpreter_client';
 
-type AppConfigStyle = AppConfig & { terminal?: boolean | 'auto'; docked?: boolean | 'docked' };
-
 const logger = getLogger('py-terminal');
 
-const validate = (config: AppConfigStyle, name: string, default_: string) => {
-    const value = config[name] as undefined | boolean | string;
-    if (value !== undefined && value !== true && value !== false && value !== default_) {
-        const got = JSON.stringify(value);
-        throw new UserError(
-            ErrorCode.BAD_CONFIG,
-            `Invalid value for config.${name}: the only accepted` +
-                `values are true, false and "${default_}", got "${got}".`,
-        );
-    }
-    if (value === undefined) {
-        config[name] = default_;
-    }
+// Configuration options for this plugin go here:
+const terminal_settings = checkedConfigOption({ possible_values: [true, false, 'auto'], default: 'auto' });
+const docked_settings = checkedConfigOption({ possible_values: [true, false, 'docked'], default: 'docked' });
+
+type AppConfigStyle = AppConfig & {
+    terminal?: (typeof terminal_settings.possible_values)[number];
+    docked?: (typeof docked_settings.possible_values)[number];
```

Comment:
I'm not really sure to understand how it works. What is the `[number]` doing here?

Reply:
It narrows the type to *members* of the array. As in "the type of an object when the preceding array is index by a number".

To take an example: `(typeof terminal_settings.possible_values)` is the type of the whole array; `(typeof terminal_settings.possible_values)[number]` is the type of a single value of that array.

Reply:
I'm not sure whether I'm glad to know this trick or horrified by it now 😂